### PR TITLE
Payment nudges

### PR DESCRIPTION
**Pull request by timuric**
While going through the tutorial, I added some optimizations to the content:

* Linked chapters together
* Added prerequisite reading as it is better that the reader first understands the payment APIs usage before building them. This removes the additional context required
* Made some sentences to be more concise 
* Change slightly the order of the content for fluidity

Next step suggestions:
We can extract the next parts and present them as a separate repo for reference, as most of the tutorial can be technology agnostic, making it appealing to a wider audience. The next js parts are quite small so we can leave out specifics in the README of the repository.

### CODE REVIEW

File: docs/developer/extending/apps/building-payment-app.mdx [hunk 1]
```diff
@@ -4,158 +4,73 @@ title: How to Build a Payment App
...
-
-With our transaction events mapped out, let's determine which webhooks we need to implement to handle these events.
-
-When exploring the [Transaction Events](developer/extending/webhooks/synchronous-events/transaction.mdx) documentation, you may notice that three webhooks can result in the `CHARGE_SUCCESS` event: [`TRANSACTION_CHARGE_REQUESTED`](developer/extending/webhooks/synchronous-events/transaction.mdx#transaction-charge), [`TRANSACTION_INITIALIZE_SESSION`](developer/extending/webhooks/synchronous-events/transaction.mdx#initialize-transaction-session) and [`TRANSACTION_PROCESS_SESSION`](developer/extending/webhooks/synchronous-events/transaction.mdx#process-transaction-session).
-
-There are two reasons for why **we are not going** with the `TRANSACTION_CHARGE_REQUESTED` webhook:
-
-1. It is designed for the staff users to charge the payment manually. Since we want a checkout user (possibly unauthenticated) to be able to pay for the order, it is not a good fit.
-2. It does not allow the return of the `CHARGE_ACTION_REQUIRED` event status.
+### Prerequisites
```
Comment (peelar) [inline docs/developer/extending/apps/building-payment-app.mdx:17]: **suggestion (non-blocking)**: Right now, we have two "prerequisites" sections. The latter one also contains "knowledge" prerequisites:

> - Basic understanding of [Saleor Apps](developer/extending/apps/overview.mdx)
> - Familiarity with [Next.js](https://nextjs.org/)

I suggest either merging those two into this "knowledge prerequisites" section or look for a different name for those sections.

File: docs/developer/extending/apps/building-payment-app.mdx [hunk 2]
```diff
@@ -4,158 +4,73 @@ title: How to Build a Payment App
...
 
-`TRANSACTION_PROCESS_SESSION` is also not a match. It is triggered when `transactionProcess` mutation is called, but only when the `TRANSACTION_INITIALIZE_SESSION` was called first and returned `CHARGE_ACTION_REQUIRED` or `AUTHORIZE_ACTION_REQUIRED` event status. This webhook will be useful for the second step of our payment process.
+- Understanding of [Payment Apps Usage](/developer/payments/payment-apps)
+- Understanding of [Transactions](/developer/payments/transactions.mdx)
 
-That leaves us with the `TRANSACTION_INITIALIZE_SESSION` webhook. It is triggered on the [`transactionInitialize`](api-reference/payments/mutations/transaction-initialize.mdx) mutation which can be used without any permissions (unless you are using [the `action` field](api-reference/payments/mutations/transaction-initialize.mdx#transactioninitializeactiontransactionflowstrategyenum-)). It can result in the `CHARGE_ACTION_REQUIRED` event status, as required.
+### What will we build
```
Comment (peelar) [inline docs/developer/extending/apps/building-payment-app.mdx:22]: **thought**: We should decide whether we follow [The Chicago Manual of Style](https://en.wikipedia.org/wiki/The_Chicago_Manual_of_Style) only in titles or all across the board. Maybe @patrys has an opinion on it as he was the original initiator.

**Review by peelar — DISMISSED** [review]

File: docs/developer/extending/apps/building-payment-app.mdx (hunk 2 above)
Comment (patrys) [reply docs/developer/extending/apps/building-payment-app.mdx:22]: I'd love for it to be enforced all across the board, preferably by a linter and an auto-formatter. On the PR scale, I think consistency within a single page is preferable.

Comment (timuric) [reply docs/developer/extending/apps/building-payment-app.mdx:22]: Will be fixed by
https://github.com/saleor/saleor-docs/pull/1258

**Review by krzysztofwolski — APPROVED** [review]

File: docs/developer/extending/apps/building-payment-app.mdx (hunk 2 above)
Comment (timuric) [reply docs/developer/extending/apps/building-payment-app.mdx]: There is also a transformation to achieve consistency quickly, with few limitations. Unfortunately `eslint -fix` can't help here
https://github.com/saleor/saleor-docs/pull/1257

**Pull request merged**